### ISSUE TITLE
Increasing MOF file size limit? 'MOF file size exceeded max size limit'

### ISSUE THREAD

**Issue by garrytrinder** [microsoft/Microsoft365DSC#357]
After compiling the exported configuration and increasing the MaxEnvelopeSize on my machine I get an error. 

The compiled MOF file is 12Mb, is there a way to increase the MOF file size limit?

```
PS C:\WINDOWS\system32> Set-Item -Path WSMan:\localhost\MaxEnvelopeSizeKb -Value 16384
PS C:\WINDOWS\system32> Start-DscConfiguration -Path C:\dsc\O365TenantConfig\ -Wait -Force
MOF file size exceeded max size limit.
    + CategoryInfo          : ResourceBusy: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : MI RESULT 27
    + PSComputerName        : localhost
```

**Comment by andikrueger** [comment]
The SharePointDSC wiki offers an article about increasing the size limit: https://github.com/dsccommunity/SharePointDsc/wiki/Error-Exceeded-the-configured-MaxEnvelopeSize-quota

Which didn't work. Did you run the command in an elevated PowerShell session?

Your setting is within the suspported range. Max value is 1039440. [source](https://docs.microsoft.com/de-de/windows/win32/winrm/installation-and-configuration-for-windows-remote-management?redirectedfrom=MSDN#protocol-default-settings)

**Comment by garrytrinder** [comment]
Thanks for response @andikrueger 

Yep, all run in an elevated session, the article you linked to worked but then I hit this issue, see below.

![image](https://user-images.githubusercontent.com/11563347/73942782-90ab1d80-48e7-11ea-817f-5358f2a72130.png)

**Comment by NikCharlebois** [comment]
Couple of questions:

1 - What is the size of the compiled .MOF?
2 - Is the plan to use the machine where you ran Start-DSCConfiguration from as the 'consistency agent'? Meaning, is this a server that will be up all the time and do the 15 minutes consistency checks or are you just trying to deploy the config and not care about the consistency checks (monitoring of drifts)? Reason I am asking this is that if you want consistency, then splitting your configuration across multiple consistency agents might be another alternative.

**Comment by garrytrinder** [comment]
The size of the compiled MOF file is 12Mb.

I don’t really have a plan, at the moment I’m just testing out the capabilities of this module & understanding the setup as well as any limitations.

I expected that I would be able to export the configuration from a tenant and then use it to either check consistency or deploy changes in my tenant directly but the generated configuration seems to be too big and has hit some kind of upper limit.

**Comment by NikCharlebois** [comment]
@garrytrinder I've followed-up with the DSC team and there is indeed a file size limit that the LCM can handle. The recommendation for large DSC files would be to break it down into separate smaller files (say EXO.ps1, SPO.ps1, SC.ps1, etc.) and use the partial configuration approach (Publish-DSCConfiguration) to handle this. Let me know if you'd need additional information on how to proceed with this. Thanks

**Comment by garrytrinder** [comment]
Could you consider this an enhancement to the `Export` cmdlet? It would be nice to have the configuration broken down into separate files, say one per workload, so that it doesn't need to be manually broken down.

**Comment by NikCharlebois** [comment]
Decision taken not to embed this inside of M365DSC. We want to make sure users understand DSC at its core before using it. Customers can arrange to break down the export into multiple files by running multiple export processes in parallel.
